cloud/api-gateway/src/main/java/com/qpay/apigateway/filter/AuthenticationGatewayFilterFactory.java

```diff
@@ -1,57 +1,63 @@
 package com.qpay.apigateway.filter;
 
-import com.qpay.apigateway.exception.MissingTokenException;
-import com.qpay.apigateway.utility.PathsUtils;
+import com.qpay.apigateway.client.AuthClient;
+import com.qpay.apigateway.exception.AuthenticationException;
+import com.qpay.apigateway.exception.WrongPathException;
 import org.springframework.cloud.gateway.filter.GatewayFilter;
 import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory;
-import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpHeaders;
 import org.springframework.http.server.reactive.ServerHttpRequest;
 import org.springframework.stereotype.Component;
-import org.springframework.web.reactive.function.client.WebClient;
-import org.springframework.web.server.ServerWebExchange;
-import reactor.core.publisher.Mono;
+
+import java.util.HashSet;
+import java.util.Set;
 
 @Component
 public class AuthenticationGatewayFilterFactory extends
         AbstractGatewayFilterFactory<AuthenticationGatewayFilterFactory.Config> {
 
-    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private final AuthClient authClient;
+
+    private final Set<String> allowedPaths = new HashSet<>(
+            Set.of("/api/v1/customers",
+                    "/api/v1/merchants",
+                    "/api/v1/wallets",
+                    "/api/v1/payments",
+                    "/api/v1/qrcode",
+                    "/api/v1/auth"));
 
-    private final WebClient webClient;
 
-    public AuthenticationGatewayFilterFactory(final WebClient webClient) {
+    public AuthenticationGatewayFilterFactory(final AuthClient authClient) {
         super(Config.class);
-        this.webClient = webClient;
+        this.authClient = authClient;
     }
 
     @Override
     public GatewayFilter apply(final Config config) {
         return (exchange, chain) -> {
             final var request = exchange.getRequest();
-            if (authMissing(request)) {
-                return onError(exchange, HttpStatus.UNAUTHORIZED);
+            final var path = request.getPath().toString();
+            if (isPathAllowed(path)) {
+                throw new WrongPathException();
+            }
+            if (authHeaderExists(request)) {
+                throw new AuthenticationException();
```

Comment:
I've already suggested it in previous comment
```suggestion
    throw new AuthorizationException("Missing authorization header");
```